### DISCUSSION TITLE
v5.0

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#11883] [General] [answered]
![Downloads](https://img.shields.io/github/downloads/gitextensions/gitextensions/v5.0/total.svg)

## Release Notes Highlights

* Requires: **.NET 8.0 Desktop Runtime [v8.0.8](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-8.0.8-windows-x64-installer) or later**
*  Git Extensions is now a native **64bit application** and installs under "C:\Program Files\GitExtensions" instead of "C:\Program Files (x86)\GitExtensions".
* Git Extensions extensibility API updated
    * The extensibility API surface updated to use interfaces instead of concrete types (e.g., `IGitModule` instead of `GitModule`). Plugins must now reference `GitExtensions.Extensibility` (instead of `GitUI`, etc.). See https://github.com/gitextensions/gitextensions.pluginmanager/pull/73 for reference implementation.
    * Dependency injection: `IGitUICommands` now implements `IServiceProvider` and provides service resolution.
* Support `git diff` colouring
* Support search in repository with `git grep`
* Multiple improvements for handling user inputs for custom scripts - e.g., multiple inputs, input labelling, default values, scripts invocation for selected files, support arbitrary icons for scripts.
* Performance improvements
* UI and usability improvements
* WSL-related improvements
* Add support of [Difftastic](https://difftastic.wilfred.me.uk/) as diff viewer

## What's Changed

* CI: more meaningful icon instead of displaying a circle by @pmiossec in https://github.com/gitextensions/gitextensions/pull/10893
* CI: allow click on status cell to open build report by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11487
* Script user input improvement by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11281
* Add Revision Graph option to fan out all children lanes directly at parent by @mstv in https://github.com/gitextensions/gitextensions/pull/11327
* Render graph lines with less curves by @mstv in https://github.com/gitextensions/gitextensions/pull/10637
* Form Rebase: Add rebase dependent branches `--update-refs` rebase option by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11335
* git-grep UI by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11350
* FileViewer: ShowSyntaxHighlighting hotkey and context menu by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11530
* Add a default shortcut for "CreateSquashCommit" by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11585
* Form commit: Improve commit templates menu by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11584
* Git diff coloring by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11590
* FormResolveConflicts: Improve help when merging/rebasing by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11619
* feat(FormCommit): Add Conventional Commit message support by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11665
* Allow user scripts to operate on selected files by @SlugFiller in https://github.com/gitextensions/gitextensions/pull/11239
* Difftastic as diff viewer by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11677
* Allow selecting custom icons for user defined scripts by @mdonatas in https://github.com/gitextensions/gitextensions/pull/11709
* Diff moved lines: Use dimmed-zebra algorithm by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11721
* Improve recovery of files UX by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11727
* Improve recovery form by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11753
* feat(ScriptOptionsProvider): {SelectedRelativePaths} by @mstv in https://github.com/gitextensions/gitextensions/pull/11773
* DeleteRemote: Add an option to delete a local tracking branch by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11661

### Fixes
* DPI-aware scaling improvements by @mdonatas
* Trigger RepoChangedNotifier after script / plugin by @mstv in https://github.com/gitextensions/gitextensions/pull/11309
* Fix failure to add repository path as safe directory by @bdukes in https://github.com/gitextensions/gitextensions/pull/11415
* Ignore fetch/push options in configuration of remote repos by @mstv in https://github.com/gitextensions/gitextensions/pull/11429
* Include default system code page in default encodings by @qgppl in https://github.com/gitextensions/gitextensions/pull/11486
* Fix avatars not loading behind a corporate proxy by @superhoang in https://github.com/gitextensions/gitextensions/pull/11596
* Open GitHub release page for portable apps instead of downloading msi by @RussKie in https://github.com/gitextensions/gitextensions/pull/11743
* fix(ShellEx): Support adding multiple files by @mstv in https://github.com/gitextensions/gitextensions/pull/11770

## New Contributors
* @snelltheta made their first contribution in https://github.com/gitextensions/gitextensions/pull/11410
* @pmgiant made their first contribution in https://github.com/gitextensions/gitextensions/pull/11432
* @bdukes made their first contribution in https://github.com/gitextensions/gitextensions/pull/11415
* @superhoang made their first contribution in https://github.com/gitextensions/gitextensions/pull/11596
* @rstm-sf made their first contribution in https://github.com/gitextensions/gitextensions/pull/11637
* @redcatH made their first contribution in https://github.com/gitextensions/gitextensions/pull/11759
* @astos-marcb made their first contribution in https://github.com/gitextensions/gitextensions/pull/11791
* @emmanuel-ferdman made their first contribution in https://github.com/gitextensions/gitextensions/pull/11793

**Full Changelog**: https://github.com/gitextensions/gitextensions/compare/v4.2.1...v5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/gitextensions/gitextensions/releases/tag/v5.0'>v5.0</a>.</em>

**Comment by chirontt** [comment]
A manual build of this v5.0 version for Windows on Arm64 is available in my fork: https://github.com/chirontt/gitextensions/releases/tag/v5.0-win32-aarch64

**Comment by SlugFiller** [comment]
Seems like a WinGet release will be delayed pending a merge of https://github.com/vedantmgoyal9/vedantmgoyal9/pull/2214

**Reply by sanghel-orbyta** [reply]
event though I'm using it heavily, I'm not familiar with winget's release lifecycle

do you know what's keeping it held up, or have an ETA?

AFAICT there's no "use preview" for the `winget install` command..
![image](https://github.com/user-attachments/assets/eea486ac-76aa-4808-814f-f718d2560575)

**Reply by pmiossec** [reply]
A PR to add the v5.0 package is required:
* No manifest available for v5.0: https://github.com/microsoft/winget-pkgs/tree/master/manifests/g/GitExtensionsTeam/GitExtensions
* Still no PR to change that: https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+gitextensions

I think that any contribution is welcomed by winget project...

**Reply by sanghel-orbyta** [reply]
FWIW, I've created [the PR for the 5.0 upgrade manifest](https://github.com/microsoft/winget-pkgs/pull/179184)...

I've stupidly done it manually (in a codespace), having later seen that previous manifests were create with the winget CLI...

I hope I haven't created any issues for the team... just trying to be useful.

**Reply by sanghel-orbyta** [reply]
Guh, my PR is being blocked due to a URL in the manifest not passing validation...

I've created the issue  #11966, it's over my head...

**Reply by SlugFiller** [reply]
WinGet PRs for packages with GitHub releases, are automatically generated using a script. Or rather, have been. But that script has been revamped, with only a small percentage of the packages converted.

The PR I linked updates that script to make PRs for GitExtensions releases. But it, and similar PRs for many other packages, have yet to be merged.

**Reply by namtab00** [reply]
If anyone is following this, the winget package validation pipeline fails for the PR https://github.com/microsoft/winget-pkgs/pull/179204 I've tried to contribute.

I've signaled this to the maintainers [here](https://github.com/gitextensions/gitextensions/issues/11966#issuecomment-2406094363).

**Reply by namtab00** [reply]
@SlugFiller [My PR](https://github.com/microsoft/winget-pkgs/pull/179204) for the winget v5 update has been manually validated by the winget team.

The v5 for GitExtensions should now be available via winget.

**Comment by AdamSchuttovits** [comment]
Just to save some time for people looking for dark mode, no, it's not ready yet, see #11904 . You can still use version [3.5.4](https://github.com/gitextensions/gitextensions/releases/tag/v3.5.4).

**Reply by sanghel-orbyta** [reply]
ah man, I'm anxiously scanning each release notes for that one line, but here I am "stuck" on 3.5.3 cause my 40yo eyes can take light themes anymore...

**Reply by AdamSchuttovits** [reply]
if it makes you feel better, I'm in the same situation :)

**Comment by Essisten** [comment]
Unfortunately .NET 8 is not supported on Windows 7. I guess that I will have to stay with the previous version...

**Reply by vbjay** [reply]
https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962. You are so far  beyond out of support.

**Comment by Pippen19** [comment]
I get this error in new version 5.0:
![GitExtensions_bA61mVvmHG](https://github.com/user-attachments/assets/7004277e-bbbe-4bf9-9abb-86c9d53e96fc)

What can I do?

**Reply by pmiossec** [reply, marked as answer]
You have to update the git config to change the path toward `GitExtensions.exe` because it is now in `C:/Program Files/GitExtensions`

**Reply by Pippen19** [reply]
Thank you very much for your fast answer!
